src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler =
+                    operationStrategy.getOperationHandler(transaction.getOperation());
+
+            if (handler == null) {
+                throw new IllegalArgumentException(
+                        "No handler found for operation: " + transaction.getOperation()
+                );
+            }
```

Comment:
The code checks if the `handler` is `null` and throws an `IllegalArgumentException` if no handler is found for a given operation. This is a good practice for handling unexpected cases. Ensure that the `OperationStrategy` is correctly set up to provide handlers for all expected operations to avoid this exception.